presto-main/src/main/java/com/facebook/presto/server/PluginManager.java

```diff
@@ -196,11 +198,32 @@ private void loadPlugin(URLClassLoader pluginClassLoader)
 
     public void installPlugin(Plugin plugin)
     {
-        plugin.setNodeManager(nodeManager);
-        plugin.setTypeManager(typeRegistry);
-        plugin.setBlockEncodingSerde(blockEncodingManager);
-        plugin.setPageSorter(pageSorter);
-        plugin.setPageIndexerFactory(pageIndexerFactory);
+        ConnectorFactoryContext connectorFactoryContext = new ConnectorFactoryContext()
```

Comment:
Can you make this a named static class?  It will make debugging nicer, and will not hold an implicit static reference on the `PluginManager`.